### ISSUE TITLE
Wrong DNS answer format randomly

### ISSUE THREAD

**Issue by boynux** [skynetservices/skydns#254]
Seems sometimes DNS answer sections is not correct when the FQDN is a CNAME and it's happening randomly. if I query `ncpstagingservicestorage.blob.core.windows.net.` for example several times then the answers might be one of these two formats:

    A? ncpstagingservicestorage.blob.core.windows.net. 2/0/0 blob.by4prdstr03a.store.core.windows.net. [39m31s] A 40.78.112.72,  ncpstagingservicestorage.blob.core.windows.net. [5m] CNAME blob.by4prdstr03a.store.core.windows.net.

or 

    A? ncpstagingservicestorage.blob.core.windows.net. 2/0/0 ncpstagingservicestorage.blob.core.windows.net. [5m] CNAME blob.by4prdstr03a.store.core.windows.net., blob.by4prdstr03a.store.core.windows.net. [39m31s] A 40.78.112.72

That's a TCPDUMP output, if you look the first section in answer should reflect the FQDN of query which doesn't in first output but in second one it's fine. So the client refuses to parse the answer (ruby in this case).

**Comment by miekg** [comment]
This is (I believe) #217

**Comment by boynux** [comment]
Yes, could be that. I checked that before. I think I should have mentioned that too.
The difference is that it's not the order of CNAME and A records that matters (as described in #217) but the answer FQDN is incorrect.
Potentially related bugs.

Thank you

**Comment by miekg** [comment]
Planning to fix these bugs this week.

**Comment by Dynom** [comment]
I'm having similar issues. By any chance, did you made any progress on this issue @miekg ?

**Comment by miekg** [comment]
@Dynom not yet, I've chopped away other issues in my open sources project and our now finally approaching skydns
